tests/wpunit/Tribe/Events/Aggregator/Record/AbstractTest.php

```diff
@@ -282,6 +281,69 @@ public function should_correctly_insert_posts() {
 		$this->assertEquals( 0, $activity->count( Main::POSTTYPE,'skipped' ) );
 	}
 
+	/**
+	 * It should correctly insert posts
+	 *
+	 * @test
+	 */
+	public function should_not_duplicate_when_uid_is_already_present() {
+		// Insert Manual Test as a valid UNIQUE ID field source.
+		\Tribe__Events__Aggregator__Record__Abstract::$unique_id_fields['manual-test'] = [
+			'source' => 'uid',
+			'target' => 'uid',
+		];
+
+		$post = $this->factory()->post->create_and_get( [
+			'post_type'      => Records::$post_type,
+			'post_status'    => Records::$status->schedule,
+			'post_date'      => date( 'Y-m-d H:i:s' ),
+			'post_mime_type' => 'ea/manual-test',
+		] );
+
+		$record = new Record();
+		$record->set_post( $post );
+		$record->frequency             = (object) [
+			'id'       => 'on_demand',
+			'interval' => 0,
+		];
+		$record->meta['post_status']   = 'publish';
+		$record->meta['origin']        = 'manual-test';
+		$record->meta['ids_to_import'] = 'all';
+
+		$uid = '__unique_uuid_for_manual_test__' . time();
+
+		$successful_child_record = $this->factory()->post->create( [
+			'post_type'      => Records::$post_type,
+			'post_parent'    => $record->post->ID,
+			'post_status'    => Records::$status->success,
+			'post_date'      => date( 'Y-m-d H:i:s' ),
+			'post_mime_type' => $record->post->post_mime_type,
+		] );
+
+		$event_id = $this->factory()->event->create( [
+			'meta_input' => [
+				'_uid' => $uid,
+			]
+		] );
+
+		$manual_test_record = new \Tribe__Events__Aggregator__Record__Manual_Test( $record );
+		$manual_test_record->meta['origin'] = 'manual-test';
+
+		// Enforce this particular Static variable.
+		\Tribe__Events__Aggregator__Record__Manual_Test::$unique_id_fields = \Tribe__Events__Aggregator__Record__Abstract::$unique_id_fields;
+
+		$data = [
+			[
+				'uid' => $uid,
```

Comment:
Adding, either here or in a dedicated test, a `uid` pointing to an event that is not existing, would test the mixed, and more realistic nature, of this code.